tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -1316,10 +1324,14 @@ int DuckDBPyConnection::GetRowcount() {
 }
 
 void DuckDBPyConnection::Close() {
+	if (connection) {
+		Checkpoint();
```

Comment:
We internally checkpoint when the connection goes out of scope. Calling checkpoint explicitly here means that the `disable_checkpoint_on_shutdown` method is not respected.